service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/AbstractServiceClient.java

```diff
@@ -66,6 +67,15 @@
     }
 
     protected final <T> FutureTaskResultToDomainObjectPromiseWrapper<T> execute(JavaType javaType, String uri, String httpMethod, Map<String, String> headersMap, Object requestObject, String commandName, boolean requestCachingEnabled) throws IOException {
+        return execute(javaType, uri, httpMethod, headersMap, requestObject, commandName, requestCachingEnabled, null);
```

Comment:
As splitting and merging happens at generated service client impl, ideally no change is required in AbstractServiceClient.java.